pkg/security/probe/probe.go

```diff
@@ -319,6 +323,16 @@ func (p *Probe) DispatchEvent(event *model.Event) {
 		return eventJSON, event.GetEventType(), err
 	})
 
+	// ensure that all the fields are resolved before sending
+	event.FieldHandlers.ResolveContainerID(event, &event.ContainerContext)
```

Comment:
I would call these function only when required, in the RuleMatch as before or in the LookupEventOnProfiles